modules/sdk-coin-atom/src/lib/transaction.ts

```diff
@@ -0,0 +1,193 @@
+    const outputs: TransactionRecipient[] = [];
+
+    const explanationResult: TransactionExplanation = {
+      displayOrder,
+      id: this.id,
+      outputs,
+      outputAmount: '0',
+      changeOutputs: [],
+      changeAmount: '0',
+      fee: { fee: this.atomTransaction.gasBudget.gas.toString() }, // TODO - explore the fee explanation more, do we need to show limit or base amount
+      type: this.type,
+    };
+
+    switch (this.type) {
+      case TransactionType.Send:
+        return this.explainTransferTransaction(result, explanationResult);
+      default:
+        throw new InvalidTransactionError('Transaction type not supported');
+    }
+  }
+
+  /**
+   * Set the transaction type.
+   *
+   * @param {TransactionType} transactionType The transaction type to be set.
+   */
+  transactionType(transactionType: TransactionType): void {
+    this._type = transactionType;
+  }
+
+  /**
+   * Sets this transaction payload
+   *
+   * @param rawTransaction
+   */
+  fromRawTransaction(rawTransaction: string): void {
+    try {
+      this._atomTransaction = Transaction.deserializeAtomTransaction(rawTransaction);
+      this._type = TransactionType.Send;
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  serialize(): string {
+    // TODO BG-67032 - if possible use Stargate Client builder to build [and sign, if applicable] tx using data types from this tx class
+    // Needs research: how to get stargate client builder without feeding an OfflineSigner private key
+    throw new Error('not yet implemented');
+  }
+
+  static deserializeAtomTransaction(txHex: string): AtomTransaction {
+    const decodedTx = decodeTxRaw(fromBase64(txHex));
+    const registry = new Registry(defaultRegistryTypes);
+    // TODO - support many messages decoding, right now we are just taking one message for send tx type
+    const encodedMessages = decodedTx.body.messages as EncodeObject[];
+    const decodedMessage = registry.decode(encodedMessages[0]); // TODO - do we need this?
+    let type: AtomTransactionType;
+    if (encodedMessages[0].typeUrl === sendMsgType) {
+      type = AtomTransactionType.Pay;
+    } else {
+      throw new Error('Transaction type not supported: ' + encodedMessages[0].typeUrl);
+    }
+    const sendMessageData: MessageData = {
+      typeUrl: decodedMessage.typeUrl,
+      value: {
+        fromAddress: decodedMessage.fromAddress,
+        toAddress: decodedMessage.toAddress,
+        amount: decodedMessage.amount,
+      },
+    };
+    const sequence = Number(decodedTx.authInfo.signerInfos[0].sequence);
+    const explicitSignerData: SignerData = {
+      accountNumber: 100, // TODO - get this from tx data somehow
+      sequence,
+      chainId: thetaTestnetChainId, // TODO - figure out if we can get this from tx data
+    };
+    const gasBudget: GasFeeLimitData = {
+      amount: decodedTx.authInfo.fee?.amount as Coin[],
+      gas: Number(decodedTx.authInfo.fee?.gasLimit),
```

Comment:
Similar to the comment in the interface file

Reply:
From my comment before, it was coming back as a Long, so I cast it to a number type, but we can keep it as a long or even convert to a string if we wish.